sys/key_value.js

```diff
@@ -75,11 +78,22 @@ class KVBucket {
         return hyper.get(storeReq).then((dbResult) => {
             if (dbResult.body && dbResult.body.items && dbResult.body.items.length) {
                 const row = dbResult.body.items[0];
-                return {
+                const result = {
                     status: 200,
                     headers: row.headers,
                     body: row.value
                 };
+
+                if (isNoCacheRequest && isUnmodifiedSinceRequest) {
```

Comment:
IF we are not looking at the information in row to decide whether we should throw, why can't we do this in line 63 already?

Reply:
Because we need to fetch data from cassandra to compare the result timestamp with the value from the header. If the request is just a no-cache request we avoid the fetching part and just return 404 so the logic on each endpoint fetches and stores the latest (as its already implemented).

Reply:
Also typo. Fixed now